From d00e7037ebb7cc4679152f776e8af497c49498ae Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 20 Sep 2024 08:24:37 +0300
Subject: [PATCH] [FSW-9512] fs1: separate pre-init constants

Persistent DCCM(P-DCCM) will be used by different
variables which are 3 types.
1. Variables which hold 0 by default, Those
doesnt has any issue because P-DCCM will be
written with 0 by BOOTROM during cold reset.
2. Variables which hold constant other than 0,
and which never modified in run time. This
values can be initialized in any reset.
3. Variables which hold pre-init values, and
those are modified during run time and required
for next non-cold reset. This values need to
initialize only in cold reset.

Change-Id: Id6a4a06babd0133f3b5d99d24ea23f31ce67c08b
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index f5dbd8b..37632bf 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -18,7 +18,6 @@
 	.recovery_flag = 0,
 	.force_recovery = 0,
 	.validation_pass = 0,
-	.valid_manifest_idx = 0,
 	.fw_upd_switch_manifest = 0,
 	.efuse_read = 0,
 	.curr_img = 0,
diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 51d19e7..73532a7 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -72,12 +72,17 @@
 	return (boot_rom_global.die_type == FS1_IO_DIE_MASTER);
 }
 
-static void boot_rom_persist_data_non_zero_init(void)
+static void boot_rom_global_pre_init_values(void)
 {
-	/* rom_persistent_section default values */
+	/*
+	 * rom_persistent_section pre-init values.
+	 * Only assign during cold reset.
+	 * Below values modified during run time and used in next non-cold reset cases.
+	*/
 	boot_rom_global.die_id = FS1_INVALID_DIE_ID;
 	boot_rom_global.die_type = FS1_INVALID_DIE_TYPE;
 	boot_rom_global.sku = FS1_INVALID_SKU;
+	boot_rom_global.valid_manifest_idx = HL_FW_SPI_MANIFEST_MAX_NUM;
 }
 
 #define HL_ARC1_DCCM_PERSIST_SECTION_SIZE		(3 * 1024) /* 3KB */
@@ -99,11 +104,18 @@
 	cold_rst_data.arc1_persist_dccm_init_done = 1;
 	regmap_writel(cold_rst_data.data, IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
 
+	boot_rom_global_pre_init_values();
+
 non_zero_init:
 	/* Non zero initializers only required for master(die 0) */
-	if (die_id == FS1_IO_DIE_0)
-		boot_rom_persist_data_non_zero_init();
+	if (die_id == FS1_IO_DIE_0) {
+		/*
+		 * Assign pre-init constants which never modified during run time.
+		 * So, upon any reset safe to re-init them.
+		 * For example, manifest constants.
+		 */
 		lfs_globals_pre_init();
+	}
 }
 
 void boot_rom_clear_status(void)
